### PR TITLE
create and truncate instead of rename.

### LINKED ISSUE #12: panic: rename [temp location] [dest]: invalid cross-device link
This problem has been recently introduced, a build from december (commit 977de34db80e88f0981d9289ff6f3ad7fee8640d) didn't have this issue.

### PR DESCRIPTION
**Pull request by mwhooker**
Resolves #12

### CODE REVIEW

**Comment by mwhooker** [comment]
This undoes work in #5 which was supposed to help with concurrent downloads. Let's hold off until we can be sure we won't see those errors

**Comment by mwhooker** [comment]
I tested with this script and didn't see any issues

```sh
#/usr/bin/env bash
set -x

./mhook --bucket wercker-development --project sentcli --branch master --commit latest linux_amd64/sentcli $(pwd)/download/2 &
./mhook --bucket wercker-development --project sentcli --branch master --commit latest linux_amd64/sentcli $(pwd)/download/2 &
./mhook --bucket wercker-development --project sentcli --branch master --commit latest linux_amd64/sentcli $(pwd)/download/2
```

File: glide.yaml [hunk 1]
```diff
@@ -2,6 +2,6 @@ package: main
 
 import:
   - package: github.com/codegangsta/cli
-  - package: github.com/crowdmob/goamz/aws
-  - package: github.com/crowdmob/goamz/s3
+  - package: github.com/AdRoll/goamz/aws
+  - package: github.com/AdRoll/goamz/s3
```
Comment (termie) [inline glide.yaml]: what was the motivation for the change here?

**Comment by termie** [comment]
lgtm, would prefer keeping the same version of goamz as our other software if there isn't anything required in adroll

**Comment by mwhooker** [comment]
the goamz repo is deprecated. we should move all our stuff over

**Comment by hatchan** [comment]
Maybe it's time to migrate to the official AWS go library?

**Comment by hatchan** [comment]
We could download to the same directory as the destination, but using a temp file:

```
tmpDst, err := ioutil.TempFile(filepath.Dir(opts.Destination), filepath.Base(opts.Destination))
```

Then do the rename. This prevents the cross partition issue, and the expensive copy. Plus this will prevent any issue with concurrent downloads and such.

**Comment by mwhooker** [comment]
@hatchan I do think moving to the official AWS library is a good idea down the line.

The tempfile solution seems like a good solution if there's still a problem, but I don't anticipate any, because:

a. we download kiddie-pools to different locations
b. kp will download everything to per-project directories
c. we use etags to prevent multiple downloads
d. this operation is safe: We'll never get a partially written file

**Pull request merged**